src/Mir/Trans.hs

```diff
@@ -1450,6 +1450,29 @@ initialValue (M.TyArray t size) = do
         let n = fromInteger (toInteger size)
         return $ Just $ MirExp (C.VectorRepr tp) (S.app $ E.VectorReplicate tp (S.app $ E.NatLit n) e)
       Nothing -> return Nothing
+-- TODO: disabled to workaround for a bug with muxing null and non-null refs
```

Comment:
Is there an issue for this?

Reply:
I think what we actually want to do here is clean up our handling of struct initialization so we can get rid of this `initialValue` function entirely.  I've started thinking about how to do this but don't have any concrete plans yet.  But, since the initialization cleanup likely overlaps with support for unions and untyped allocation, it will probably fit as part of our next bit of memory model work.